### ISSUE TITLE
Records persist after app uninstalled.

### ISSUE THREAD

**Issue by mtg5014** [satyan/sugar#714]
Hello.

I am having an issue on the device I am testing with, where no matter what I do, two records in my table mysteriously reappear.

I have tried temporarily inserting a line....
context.deleteDatabase("DATABASENAME"); 
...in the onStart() method of my main activity, and I have also tried this line...
MyEntity.delteAll(MyEntity.class);

Somehow, my device (an LG G5 running Marshmallow) has 2 entries in the database table that just won't go away.  

What is the proper way to "re-initialize" a database after the app is uninstalled and reinstalled?  I would think that by uninstalling my app, the android OS would automatically remove the database file, but apparently that is not the case.  I have not had this issue when using SQLite directly, without any ORM.

Any help is much appreciated!

**Comment by ribertojunior** [comment]
Have you tried to change the database version?

**Comment by mtg5014** [comment]
No - I didn't try that because nothing in my db actually changed, rather the application code which is dependent on the db entries.  I have an alarm clock app, so when records are created/saved/deleted, I have corresponding code to call the AlarmManager interface.   

I actually resolved this last night by power cycling my phone after deleting the two db entries.  Strange...but  I suspect I created some sort of one-off condition during my development iterations.  Now if I install the app, create some entries, and then uninstall/reinstall, I get a fresh database.  This is what I expected.

I will close the issue, however I do have a curious question if anyone is interested.  How would one (hypothetically) handle adding a @NotNull field to an existing table?  One would need to add the column, then populate it, then add the constraint.  Is this something SugarDB could support?  It seems if we add a field to a SugarRecord with @NotNull annotation, it will attempt to extend the db table before a patch script would run to populate the column.
